lib/Jsrt/JsrtDebugEventObject.cpp

```diff
@@ -0,0 +1,125 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+#include "JsrtPch.h"
+#include "JsrtDebugEventObject.h"
+#include "RuntimeDebugPch.h"
+#include "screrror.h"   // For CompileScriptException
+
+JsrtDebugEventObject::JsrtDebugEventObject(Js::ScriptContext *scriptContext)
+{
+    Assert(scriptContext != nullptr);
+    this->scriptContext = scriptContext;
+    this->eventDataObject = scriptContext->GetLibrary()->CreateObject();
+    Assert(this->eventDataObject != nullptr);
+}
+
+JsrtDebugEventObject::~JsrtDebugEventObject()
+{
+    this->eventDataObject = nullptr;
+    this->scriptContext = nullptr;
+}
+
+Js::DynamicObject* JsrtDebugEventObject::GetEventDataObject()
+{
+    return this->eventDataObject;
+}
+
+JsrtDebugDocumentManager::JsrtDebugDocumentManager(JsrtDebugManager* jsrtDebugManager) :
+    breakpointDebugDocumentDictionary(nullptr)
+{
+    Assert(jsrtDebugManager != nullptr);
+    this->jsrtDebugManager = jsrtDebugManager;
+}
+
+JsrtDebugDocumentManager::~JsrtDebugDocumentManager()
+{
+    if (this->breakpointDebugDocumentDictionary != nullptr)
+    {
+        AssertMsg(this->breakpointDebugDocumentDictionary->Count() == 0, "Should have cleared all entries by now?");
+
+        Adelete(this->jsrtDebugManager->GetDebugObjectArena(), this->breakpointDebugDocumentDictionary);
+
+        this->breakpointDebugDocumentDictionary = nullptr;
+    }
+    this->jsrtDebugManager = nullptr;
+}
+
+void JsrtDebugDocumentManager::AddDocument(UINT bpId, Js::DebugDocument * debugDocument)
+{
+    BreakpointDebugDocumentDictionary* breakpointDebugDocumentDictionary = this->GetBreakpointDictionary();
+
+    Assert(!breakpointDebugDocumentDictionary->ContainsKey(bpId));
+
+    breakpointDebugDocumentDictionary->Add(bpId, debugDocument);
+}
+
+void JsrtDebugDocumentManager::ClearDebugDocument(Js::ScriptContext * scriptContext)
+{
+    if (scriptContext != nullptr)
```

Comment:
so that means the scriptContext cannot be null as well.

---
In reply to: [62768297](https://github.com/Microsoft/ChakraCore/pull/926#discussion_r62768297) [](ancestors = 62768297,62760505)